packages/blade-old/MIGRATION.md

```diff
@@ -0,0 +1,195 @@
+- [From version 1.4.x to 2.x.x](#from-version-14x-to-2xx)
+
+
+
+# From version 1.4.x to 2.x.x
+
+## Migration guide for new blade-old icons
+
+### Migrating icons to new API
+
+
+```diff
+- import Icon from "@razorpay/blade-old";
++ import { AlertCircle } from "@razorpay/blade-old/icons";
+
+- <Icon name="alert-circle" fill="shade.950" size="medium" />
++ <AlertCircle fill="shade.950" size="medium" />
+```
+
+To make this process easier I would suggest you use vscode search panel to search all the instances of `name="` which will show you all the places you used icons, and replace them with their respective icon.
+
+
+## Changes in \<Icon />
+
+Icon component is now a generic wrapper to create new icons. 
+To add new icons to blade's icon library you can use the <Icon /> component like so: 
+
+```jsx
+// User defined custom icons
+export const MyCustomIcon = (props) => {
+  return (
+    <Icon viewBox="0 0 24 24" {...props}>
+      <path d="" />
+    </Icon>
+  );
+};
+
+// usage
+<Button icon={MyCustomIcon}>
+```
+
+## Migrating existing blade-old components
+
+Few of the internal components were using the old \<Icon /> component, thus we also had to change the APIs for them.
+
+```sh
+# migrate-icons [directory_path_glob] [...jscodeshift_options]
+migrate-icons src/App/** parser=jsx
+```
+
+### Automigration
+
+We've written a codemod to automate the process of replacing the `icon` prop with imported icons. 
+
+Simply install the blade-old package and run
+
+```sh
+// migrate-icons [directory_path_glob] [...jscodeshift_options]
+migrate-icons ./src/App/*
+```
+
+This codemod automatically changes 
+
+This
+```
+<Icon size="md" name="download" {...{}}>
+    <works />
+</Icon>
+```
+
+To this:
+
+```jsx
+import { Download } from "@razorpay/blade-old";
+
+<Download size="md" {...{}}>
+    <works />
+</Download>
+```
+
+### Limitations & Edge cases
```

Comment:
```suggestion
### Manual Steps
```

Rename this to Manual steps and then start this section with the same content as it is now and then add the manual steps at the bottom of this section itself so you'll end up with just 2 sections

* Codemod usage
* Manual Steps

Reply:
Limitations are part of the codemod section, renamed it to `Limitations & Edge cases of codemod`

We now have
Codemod to migrate to new API
Limitations & Edge cases of codemod
Manual Migration